client/client.go

```diff
@@ -785,20 +816,34 @@ func (c *client) GetTSAsync(ctx context.Context) TSFuture {
 }
 
 func (c *client) GetLocalTSAsync(ctx context.Context, dcLocation string) TSFuture {
+	return c.getLocalTSAsyncWithRetry(ctx, dcLocation, nil)
+}
+
+func (c *client) getLocalTSAsyncWithRetry(ctx context.Context, dcLocation string, bo *retry.Backoffer) TSFuture {
 	defer trace.StartRegion(ctx, "pdclient.GetLocalTSAsync").End()
 	if span := opentracing.SpanFromContext(ctx); span != nil && span.Tracer() != nil {
 		span = span.Tracer().StartSpan("pdclient.GetLocalTSAsync", opentracing.ChildOf(span.Context()))
 		defer span.Finish()
 	}
 
-	req := c.getTSORequest(ctx, dcLocation)
-	if err := c.dispatchTSORequestWithRetry(req); err != nil {
+	req := c.getTSORequest(ctx, dcLocation, bo)
+
+	if req.bo != nil {
+		// This is because caller should call `Wait` to get the error if dispatch failed.
+		if err := req.bo.ExecWithoutReset(ctx, func() error {
+			return c.dispatchTSORequestWithFastRetry(req)
+		}); err != nil {
+			req.tryDone(err)
```

Comment:
This retry might be meaningless if the `req.tryDone(err)` has been called elsewhere before retrying. We might need to drain it first.